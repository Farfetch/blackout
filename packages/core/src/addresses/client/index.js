/**
 * Addresses clients.
 *
 * @module addresses/client
 * @category Addresses
 * @subcategory Clients
 */

export { default as getPredictions } from './getPredictions';
export { default as getPredictionDetails } from './getPredictionDetails';

export { default as getAddresses } from './getAddresses';
export { default as getAddress } from './getAddress';
export { default as postAddress } from './postAddress';
export { default as putAddress } from './putAddress';
export { default as deleteAddress } from './deleteAddress';
export { default as putDefaultBillingAddress } from './putDefaultBillingAddress';
export { default as putDefaultShippingAddress } from './putDefaultShippingAddress';
export { default as getSchema } from './getSchema';
export { default as putDefaultContactAddress } from './putDefaultContactAddress';
export { default as deleteDefaultContactAddress } from './deleteDefaultContactAddress';
export { default as getDefaultContactAddress } from './getDefaultContactAddress';
