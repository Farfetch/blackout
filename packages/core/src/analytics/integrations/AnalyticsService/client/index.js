/**
 * @module AnalyticsService/client
 * @category Analytics
 * @subcategory Clients
 */

/**
 * Trackings client.
 */
export { default as postAnalytics } from './postAnalytics';
