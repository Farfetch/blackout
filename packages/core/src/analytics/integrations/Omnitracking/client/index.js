/**
 * @module Omnitracking/client
 * @category Analytics
 * @subcategory Clients
 */

/**
 * Trackings client.
 */
export { default as postTrackings } from './postTrackings';
export { default as postBatchTrackings } from './postBatchTrackings';
export * from './postTrackings';
