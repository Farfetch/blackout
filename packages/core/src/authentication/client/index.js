/**
 * Authentication clients.
 *
 * @module authentication/client
 * @category Authentication
 * @subcategory Clients
 */

export { default as postLogin } from './postLogin';
export { default as postLogout } from './postLogout';
export { default as postPasswordChange } from './postPasswordChange';
export { default as postPasswordRecover } from './postPasswordRecover';
export { default as postPasswordReset } from './postPasswordReset';
export { default as postRefreshToken } from './postRefreshToken';
export { default as postRefreshEmailToken } from './postRefreshEmailToken';
export { default as postRegister } from './postRegister';
export { default as postValidateEmail } from './postValidateEmail';
export { default as postUserToken } from './postUserToken';
export { default as postGuestTokens } from './postGuestTokens';
export { default as postTokens } from './postTokens';
export { default as deleteUserToken } from './deleteUserToken';
export { default as deleteGuestTokens } from './deleteGuestTokens';
export { default as deleteTokens } from './deleteTokens';
