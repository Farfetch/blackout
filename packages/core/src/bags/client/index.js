/**
 * Bags clients.
 *
 * @module bags/client
 * @category Bags
 * @subcategory Clients
 */
import { warnDeprecatedMethod } from '../../helpers';

warnDeprecatedMethod(
  '@farfetch/blackout-core',
  '@farfetch/blackout-core/bags/client',
  '@farfetch/blackout-core/bags',
);

export { default as getBag } from './getBag';
export { default as getBagOperation } from './getBagOperation';
export { default as getBagOperations } from './getBagOperations';
export { default as deleteBagItem } from './deleteBagItem';
export { default as patchBagItem } from './patchBagItem';
export { default as postBagItem } from './postBagItem';
