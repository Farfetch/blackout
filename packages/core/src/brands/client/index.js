import { warnDeprecatedMethod } from '../../helpers';

warnDeprecatedMethod(
  '@farfetch/blackout-core',
  '@farfetch/blackout-core/brands/client',
  '@farfetch/blackout-core/brands',
);

/**
 * Brands clients.
 *
 * @module brands/client
 * @category Brands
 * @subcategory Clients
 */

export { default as getBrand } from './getBrand';
export { default as getBrands } from './getBrands';
