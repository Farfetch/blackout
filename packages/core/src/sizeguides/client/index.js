/**
 * Sizeguides clients.
 *
 * @module sizeGuides/client
 * @category SizeGuides
 * @subcategory Clients
 */
import { warnDeprecatedMethod } from '../../helpers';

warnDeprecatedMethod(
  '@farfetch/blackout-core',
  '@farfetch/blackout-core/sizeguides/client',
  '@farfetch/blackout-core/sizeGuides',
);

export { default as getSizeguides } from './getSizeguides';
