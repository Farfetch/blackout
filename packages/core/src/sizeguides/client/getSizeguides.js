import client, { adaptError } from '../../helpers/client';
import join from 'proper-url-join';

/**
 * @typedef {object} GetSizeGuidesQuery
 *
 * @alias GetSizeGuidesQuery
 * @memberof module:sizeGuides/client
 *
 * @property {Array} [brandIds] - Brand ids to search for sizeguides.
 * @property {Array} [categoryIds] - Category ids to search for sizeguides.
 */

/**
 * Method responsible for loading the possible sizeguides for specific brand ids
 * and category ids.
 * This sizeguides logic should be used in cases that the project does not have
 * a category tree.
 * If your project has a category tree you should use the sizeguides logic
 * from @farfetch/blackout-core/products/details/client.
 *
 * @function getSizeGuides
 * @memberof module:sizeGuides/client
 *
 * @param {GetSizeGuidesQuery} [query] - Query parameters to apply to the
 * request.
 * @param {object} [config] - Custom configurations to send to the client
 * instance (axios).
 *
 * @returns {Promise} Promise that will be resolved when the call to the
 * endpoint finishes.
 */
export default (query, config) =>
  client
    .get(
      join('/commerce/v1/sizeguides', {
        query,
      }),
      config,
    )
    .then(response => response.data)
    .catch(error => {
      throw adaptError(error);
    });
