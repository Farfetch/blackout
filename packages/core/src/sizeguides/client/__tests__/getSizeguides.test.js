import { getSizeguides } from '..';
import {
  mockBrandId,
  mockCategoryId,
  mockQuery,
  mockSizeguides,
} from 'tests/__fixtures__/sizeguides';
import client from '../../../helpers/client';
import fixtures from '../__fixtures__/getSizeguides.fixtures';
import moxios from 'moxios';

describe('getProductSizeguides', () => {
  const expectedConfig = undefined;
  const spy = jest.spyOn(client, 'get');

  beforeEach(() => {
    moxios.install(client);
    jest.clearAllMocks();
  });

  afterEach(() => moxios.uninstall(client));

  it('should handle a client request successfully', async () => {
    const response = mockSizeguides;

    fixtures.success({
      query: mockQuery,
      response,
    });

    await expect(getSizeguides(mockQuery)).resolves.toBe(response);
    expect(spy).toHaveBeenCalledWith(
      `/commerce/v1/sizeguides?brandIds=${mockBrandId}&categoryIds=${mockCategoryId}`,
      expectedConfig,
    );
  });

  it('should receive a client request error', async () => {
    fixtures.failure({
      query: mockQuery,
    });

    await expect(getSizeguides(mockQuery)).rejects.toMatchSnapshot();
    expect(spy).toHaveBeenCalledWith(
      `/commerce/v1/sizeguides?brandIds=${mockBrandId}&categoryIds=${mockCategoryId}`,
      expectedConfig,
    );
  });
});
