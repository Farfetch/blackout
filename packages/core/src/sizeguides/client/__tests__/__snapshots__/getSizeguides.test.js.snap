// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getProductSizeguides should receive a client request error 1`] = `
Object {
  "code": -1,
  "message": "stub error",
  "status": 404,
}
`;
